CMakeLists.txt

```diff
@@ -50,6 +50,9 @@ OPTION( CURL			"Use curl for HTTP if available" ON)
 OPTION( USE_EXT_HTTP_PARSER		"Use system HTTP_Parser if available" ON)
 OPTION( DEBUG_POOL			"Enable debug pool allocator"			OFF )
 OPTION( ENABLE_WERROR			"Enable compilation with -Werror"		OFF )
+IF (UNIX AND NOT APPLE)
+	OPTION( ENABLE_REPRODUCIBLE_BUILDS	"Enable reproducible builds" 			OFF )
+ENDIF()
```

Comment:
Why `UNIX AND NOT APPLE`?  Are these GNU-only settings?  If so, what about (say) FreeBSD?  I wonder if there's a better way to detect GNUness...

Reply:
Those options are not available on all implementations of "ar" and "ranlib", unfortunately. I think being GNU is not even sufficient here, as those options were introduced not that long in the past. I even think Ubuntu 14.04 does not have the ability to build it like this.

So originally, I intended to do implement this mode as the default, such that all builds are deterministic. But as I saw that it wasn't available on quite a lot of platforms, I simply made it an option such that the distributor can decide for himself if he needs reproducible builds or not. And in case he knows what a reproducible build is and what it is for, he probably also has enough knowledge to fix his toolchain.

So I bet that some BSDs have the ability to have reproducible builds. I'd at least expect OpenBSD to have them, regarding their focus on security. But as far as I know, this is not at all available on macOS. So yeah, we _could_ probably come up with something which just tests if those tools support the required flags. But in the end, I don't think it's really gaining us much, as it is only an option for experts who know what they are doing.

Reply:
I see.  That seems reasonable...